### PR TITLE
Migrate to Upjet and Upgrade to Latest Terraform Provider

### PR DESCRIPTION
**Pull request by Henni**
### Description of your changes

brings the provider up to date with recent changes, both on the Crossplane and MongoDB Atlas side.
- migrated from terrajet to upjet
- upgraded terraform version to 1.4.9
- upgraded terraform provider version to 1.9.0

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
manually tested with a set of ~10 core resources to not have any regressions.

### CODE REVIEW

**Comment by Henni** [comment]
`run reviewable` fails because of MacOS date formatting I think
```can't run linter goanalysis_metalinter: gofmt: can't extract issues from gofmt diff output [...] parsing time "2023-07-25 23:11:59" as "2006-01-02 15:04:05 -0700": cannot parse "" as "-0700"```

**Comment by Henni** [comment]
I recommend filtering out the two "regenerate" commits during code review.

File: config/external_name.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+package config
+
+import "github.com/upbound/upjet/pkg/config"
+
+// ExternalNameConfigs contains all external name configurations for this
+// provider.
+var ExternalNameConfigs = map[string]config.ExternalName{
+	// Import requires using a randomly generated ID from provider: nl-2e21sda
+	"null_resource": config.IdentifierFromProvider,
```
Comment (haarchri) [inline config/external_name.go:13]: do you need `null_resource`?

**Review by haarchri — COMMENTED** [review]
@Henni  thanks for switching to upjet - a few small hints ;)

File: cmd/provider/main.go [hunk 2]
```diff
@@ -61,16 +64,28 @@ func main() {
...
+		Provider: config.GetProvider(),
+		// use the following WorkspaceStoreOption to enable the shared gRPC mode
+		// terraform.WithProviderRunner(terraform.NewSharedProvider(log, os.Getenv("TERRAFORM_NATIVE_PROVIDER_PATH"), terraform.WithNativeProviderArgs("-debuggable")))
+		WorkspaceStore: terraform.NewWorkspaceStore(log),
+		SetupFn:        clients.TerraformSetupBuilder(*terraformVersion, *providerSource, *providerVersion),
+	}
+
 	kingpin.FatalIfError(apis.AddToScheme(mgr.GetScheme()), "Cannot add MongoDBAtlas APIs to scheme")
-	kingpin.FatalIfError(controller.Setup(mgr, log, rl, setup, ws, pconfig.GetProvider(tf.Provider().ResourcesMap), 1), "Cannot setup MongoDBAtlas controllers")
+	kingpin.FatalIfError(controller.Setup(mgr, o), "Cannot setup MongoDBAtlas controllers")
```
[the remark this answers is not in the document]
Comment (Henni) [reply cmd/provider/main.go:89]: What impact does it have to set the managementPolicy flag? Does my providers need to support this explicitly elsewhere or is it just plug-and-play?

File: config/external_name.go (hunk 1 above)
Comment (Henni) [reply config/external_name.go:13]: I think I can just remove the whole file as the old overrides.go provides the externalName configuration for all resources anyway.

File: cmd/provider/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (haarchri) [reply cmd/provider/main.go:121]: Plug and Play with latest upjet and crossplane-runtime - check in crossplane for granular management policies

[the remark this answers is not in the document]
Comment (Henni) [reply cmd/provider/main.go:127]: done

**Review by haarchri — APPROVED** [review]
thanks looks good

**Pull request merged**